### PR TITLE
test: add empty fallback tests for ComparePage

### LINKED ISSUE #2793: test(ComparePage-empty-fallback): verify Edge Cases & Empty/Missing Inputs Verification (Variation 1)
## 🧩 Background
The component/utility at `app/compare/page.tsx` needs isolated unit and integration testing. This issue focuses specifically on **Edge Cases & Empty/Missing Inputs Verification**. Handling missing options, unconfigured objects, or empty arrays is critical to guarantee component stability in production.

## 🎯 Objective
Create the new test file `app/compare/page.empty-fallback.test.tsx` targeting Edge Cases & Empty/Missing Inputs Verification and achieve complete coverage on the specified conditions.

## 📁 Files to touch
- `app/compare/page.empty-fallback.test.tsx` (new)

## 🛠️ Implementation steps
- Render the target module or component with empty arrays or null parameters.
- Verify that a clear, non-breaking fallback UI or error message is displayed.
- Verify standard styles are maintained in this default empty layout state.
- Assert that no unexpected runtime errors or hydration failures occur.
- Check key DOM structures to make sure empty markers exist.

## ✅ Definition of done
- [ ] 5 test cases.
- [ ] `vitest run` passes.

### PR DESCRIPTION
**Pull request by Srishti-Gupta74**
## Description

Fixes #2793

Adds empty fallback and edge-case tests for `ComparePage`.

Covered scenarios:

* Safe rendering without crashes
* CompareClient rendering
* Footer rendering
* Metadata validation
* OpenGraph metadata validation

## Pillar

* [ ] 🎨 Pillar 1 — New Theme Design
* [ ] 📐 Pillar 2 — Geometric SVG Improvement
* [ ] 🕐 Pillar 3 — Timezone Logic Optimization
* [x] 🛠️ Other (Bug fix, refactoring, docs)

## Visual Preview

N/A (test-only change)

## Checklist before requesting a review:

* [x] I have read the `CONTRIBUTING.md` file.
* [x] I have tested these changes locally (`localhost:3000/api/streak?user=YOUR_USERNAME`).
* [ ] I have run `npm run format` and `npm run lint` locally and resolved all errors (CI will fail otherwise).
* [x] My commits follow the Conventional Commits format (e.g., `feat(themes): ...`, `fix(calculate): ...`).
* [ ] I have updated `README.md` if I added a new theme or URL parameter.
* [x] I have started the repo.
* [x] I have made sure that i have only one commit to merge in this PR.
* [ ] The SVG output matches the CommitPulse "premium quality" aesthetic standard (no raw elements, smooth animations, correct fonts).
* [x] (Recommended) I joined the CommitPulse Discord community for contributor discussions, mentorship, and faster PR support.

### CODE REVIEW

File: app/compare/page.empty-fallback.test.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+describe('ComparePage empty fallback behavior', () => {
+  it('renders safely without crashing', () => {
+    expect(() => render(<ComparePage />)).not.toThrow();
+  });
+
+  it('renders CompareClient content', () => {
+    render(<ComparePage />);
+
+    expect(screen.getByTestId('compare-client')).toBeTruthy();
```
[the remark this answers is not in the document]
Comment (Srishti-Gupta74) [reply app/compare/page.empty-fallback.test.tsx]: Addressed by updating the assertion and pushing the latest changes. Thanks!

File: app/compare/page.empty-fallback.test.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  });
+
+  it('renders Footer component', () => {
+    render(<ComparePage />);
+
+    expect(screen.getByTestId('footer')).toBeTruthy();
```
[the remark this answers is not in the document]
Comment (Srishti-Gupta74) [reply app/compare/page.empty-fallback.test.tsx]: Addressed in the latest commit by updating the assertions. Tests continue to pass locally.

**Review by Aamod007 — APPROVED** [review]

**Review by Aamod007 — APPROVED** [review]
